modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -197,16 +197,45 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
             {
                 CV_Assert(0 && "Internal error");
             }
-
-            for (size_t j = 2; j < dims; j++)
-                CV_Assert(inputs[0][j] == inputs[i][j]);
         }
 
         channelsMode = variableChannels ? channelsModeInput : ELTWISE_CHANNNELS_SAME;
         outputChannels = numChannels;
 
         outputs.assign(1, inputs[0]);
         outputs[0][1] = numChannels;
+
+        if (dims > 2)
+        {
+            size_t vecIdx = 0;
+            bool isVecFound = false;
+            for (size_t i = 0; i < inputs.size(); i++)
+            {
+                bool allOnes = isAllOnes(inputs[i], 2, dims);
+                if (!allOnes && !isVecFound)
+                {
+                    vecIdx = i;
+                    isVecFound = true;
```

Comment:
IMHO, such stuff should be explicitly specified through parameters instead of implicit internal handling.

Reply:
Main concern here is about correct and accurate implementation of `supportBackend()` callback.

Only OpenCV/CPU backend supports new mode (CUDA, Halide, Vulkan, even OpenCV/OpenCL doesn't support that).
Unsupported backends should be filtered out in the right way.

Reply:
@alalek, could you, please, clarify the following: if we don't change the computation core (``forward(...)`` and ``run(...)`` functions were not changed) and modify only one of the inputs' shape, expanding it in ``getMemoryShapes(...)``, we should not impact on the chosen backend. Thus, do we really need ``supportBackend()`` callback corrections here?

Reply:
`forward()` is modified below, see `if (channelsModeInput == ELTWISE_CHANNNELS_SAME && inputs[0].dims > 2)`

---

Tests filter out non-working "OpenCL" cases (as mentioned there this should not be done for OpenCV/OCL code path)

---

> modify only one of the inputs' shape

Probably we can't do that. Input is always a some output of another layer.

Reply:
@alalek, thank you for the review, added corrections for ``supportBackend()``.